content/React Query and management of server state.md

```diff
@@ -0,0 +1,75 @@
+This is a very difficult task if we try to do with Redux or the other tools that I mentioned before because we have to find a way to resolve cache management, background update and other cases that precise a little bit more of implementation when merging old and new data such as pagination or infinite scroll.
+
+## What is React Query and how it solves these problems?
+React query is a library that with 2 hooks and 1 utility (only 5kb!) provides an easy and scalable way to fetching, caching and updating asynchronous data in React.
+
+Before jump off to the code and see some examples I like to introduce how React Query models the server state.
+![React query async states](https://i.ibb.co/4M4bkQ8/Screen-Shot-2020-05-14-at-14-23-23.png)
+In the image above, we can see the different states for the server state information, let's see what is the meaning of each state:
+- Fetching: This is the initial state and occurs when the data is fetching from an outsource, typically the backend.
+- Fresh: On this state, the information is the same in both sides, this is the desired state of our application because implies that we don't need to re-fetch the information. On the other hand, this state lasts a short time because in most cases when your fetching information, this information immediately is potentially out of date.
+- Stale: Represents the data that's out of date and is used on the app currently, this is because of frontend has old information and need to re-fetch from backend, or backend is not updated because frontend didn't send the new information to backend yet. This state is particularly important as long as we want to be in the fresh state again.
+- Inactive: React Query has a garbage collector for managing cache on the browser, this state in some manner indicates on React query that if the data is not be using on the application then it's gonna be potentially deleted after some time. This is a great feature of React Query, because we want to persist the recent information for not fetching data all the time and improve the speed/UX of our interfaces.
+- Deleted: This happens when the data was inactive for a determinate time, in which case is deleted from the cache. This time can be configurable locally for each query or globally.
+
+With this approach, React Query handles the asynchronous information of our application clearly, allow us to scale and maintain our code in a better way.
+
+## Introduction to React query API
+Although React Query has only two hooks, it is highly configurable in all aspects, since the retry delay time of the queries until what is the interval time of re-fetch in background. 
```

Comment:
`Although React Query has only two hooks, it is highly configurable in all aspects, since the retry delay time of the queries until what is the interval time of re-fetch in background.`

This phrase is a little confusing, especially the end, I think what you meant is `from the retry delay time of the queries to what is the interval time of refetch in the background`, is that ok?

Reply:
Yes, maybe I'll put another example on the `to ...` sentence because it's related to the same topic and it's a little confusing